haskell-project.nix

```diff
@@ -35,10 +35,34 @@ in
       localPackagesOverlay = self: _:
         let
           fromSdist = self.buildFromCabalSdist or (builtins.trace "Your version of Nixpkgs does not support hs.buildFromCabalSdist yet." (pkg: pkg));
+          # If `root` has package.yaml, but no cabal file, generate the cabal
```

Comment:
> what seems to be best practice of:

Is it? I'm not so sure if it is a good practice. 

I don't use hpack myself, but I'd imagine that adding both `package.yaml` and the generated `.cabal` is not a great idea, since not all Haskell tools recognize hpack, and will proceed to use (the potentially outdated) `.cabal` file regardless.

(Unless you are checking for this in the pre-commit hooks and such, but not all users do that ...)

Reply:
> Is it? I'm not so sure if it is a good practice.

Yeah, I'm really not either. I used to firmly be in the "don't track the auto-generated cabal file" camp. I found the arguments in the posted link compelling though and slightly moved to the other side.

It's probably fine as is honestly.

Reply:
I think if they generate a cabal file, they should keep it up to date, so that it doesn't matter. We could generate a check for this, or check it during the package build, or let the user rely on the hpack hook in pre-commit-hooks.nix.
If the files are kept in sync like that, it seems acceptable to commit the cabal file.

Personally I like the simplicity of avoiding hpack, so this is about as much as I can add to the conversation.

Reply:
Another idea is to create a flake check that verifies that the generated (and committed) cabal file is up to date. #97